packages/web-components/src/theme/set-theme.ts

```diff
@@ -10,7 +10,15 @@ const tokenNames = Object.keys(tokens) as (keyof Theme)[];
  */
 export const setTheme = (theme: Theme) => {
   for (const t of tokenNames) {
-    document.body.style.setProperty(`--${t}`, theme[t] as string);
+    if ('registerProperty' in CSS) {
+      CSS.registerProperty({
+        name: `--${t}`,
+        inherits: true,
+        initialValue: theme[t] as string,
+      });
+    } else {
+      document.body.style.setProperty(`--${t}`, theme[t] as string);
+    }
```

Comment:
Just curious - do we know the performance delta here?

Reply:
I don't know off-hand. But let me see if I can get you an answer.

Reply:
Alright! Some initial data from a small scale test. Feel free to poke holes, or ask questions.

**Methodology**

I wrapped the function in `performance.now` calls and reported the result in the browser console, commenting out the other CSS variable registration method to get before and after timestamps. 

```js 
export const setTheme = (theme: Theme) => {
  const start = performance.now();
  for (const t of tokenNames) {
      CSS.registerProperty({
        name: `--${t}`,
        inherits: true,
        initialValue: theme[t] as string,
      });

     // document.body.style.setProperty(`--${t}`, theme[t] as string);
  }
  const end = performance.now();
  console.log(`setTheme took ${end - start}ms`);
};
```

I then ran the test at 3 different synthetic CPU profiles provided by DevTools: none, 4x, and 6x slowdown.

**Results**

| Method | No slowdown | 4x slowdown | 6x slowdown |
| --- | --- | --- | --- |
| `CSS.registerProperty` | 0.5ms | 1.8999999ms | 3ms |
| `style.setProperty` | 0.6999999ms | 2.5ms | 5.6999999ms |

The initial results of this small scale test shows a minor improvement on fast devices but a 2x improvement (but only 3ms difference) on slower devices.

Reply:
Awesome